lxd/networks_utils.go

```diff
@@ -580,3 +582,69 @@ func networkFanAddress(underlay *net.IPNet, overlay *net.IPNet) (string, string,
 
 	return fmt.Sprintf("%s/%d", ipBytes.String(), overlaySize), dev, ipStr, err
 }
+
+func networkKillDnsmasq(name string, reload bool) error {
+	// Check if we have a running dnsmasq at all
+	pidPath := shared.VarPath("networks", name, "dnsmasq.pid")
+	if !shared.PathExists(pidPath) {
+		if reload {
+			return fmt.Errorf("dnsmasq isn't running")
+		}
+
+		return nil
+	}
+
+	// Grab the PID
+	content, err := ioutil.ReadFile(pidPath)
+	if err != nil {
+		return err
+	}
+	pid := strings.TrimSpace(string(content))
+
+	// Check if the process still exists
+	if !shared.PathExists(fmt.Sprintf("/proc/%s", pid)) {
+		os.Remove(pidPath)
+
+		if reload {
+			return fmt.Errorf("dnsmasq isn't running")
+		}
+
+		return nil
+	}
+
+	// Check if it's dnsmasq
+	cmdPath, err := os.Readlink(fmt.Sprintf("/proc/%s/exe", pid))
+	if err != nil {
+		return err
+	}
+
+	// Deal with deleted paths
+	cmdName := filepath.Base(strings.Split(cmdPath, " ")[0])
+	if cmdName != "dnsmasq" {
+		if reload {
+			return fmt.Errorf("dnsmasq isn't running")
+		}
+
+		os.Remove(pidPath)
+		return nil
+	}
+
+	// Actually kill the process
+	if reload {
+		err = shared.RunCommand("kill", "-HUP", pid)
+		if err != nil {
+			return err
+		}
+
+		return nil
+	}
+
+	err = shared.RunCommand("kill", "-9", pid)
```

Comment:
Nope, will switch to it. I did look at using the Process managements function from os but those weren't a good fit, didn't think of using syscall instead.